src/main/java/net/sf/jabref/importer/fileformat/MedlinePlainImporter.java

```diff
@@ -127,91 +126,93 @@ public ParserResult importDatabase(BufferedReader reader) throws IOException {
                     }
                 }
                 String entry = current.toString();
+                if (entry.contains("-")) {
```

Comment:
You could also extract to multiple methods, `private boolean first checkLineValidity(String line)`, if false try to correct it by calling another method that tries to insert spaces or similar things.